x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/LocalModel.java

```diff
@@ -39,14 +50,17 @@
     private volatile long persistenceQuotient = 100;
     private final LongAdder currentInferenceCount;
     private final InferenceConfig inferenceConfig;
+    private final CircuitBreaker trainedModelCircuitBreaker;
+    private final AtomicLong referenceCount;
 
     public LocalModel(String modelId,
```

Comment:
We might want to make this package private. It seems that if we want to make sure that the circuit breaker stuff is handled correctly, that only things within this package should be able to ctor the model